core/husarnet.cmake

```diff
@@ -124,59 +158,59 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL Linux)
 endif()
 
 if(${CMAKE_SYSTEM_NAME} STREQUAL Windows)
-  target_link_libraries(husarnet_core iphlpapi shlwapi ws2_32)
+  target_link_libraries(${husarnet_core} iphlpapi shlwapi ws2_32)
 endif()
 
 if(${CMAKE_SYSTEM_NAME} STREQUAL Android)
-  target_link_libraries(husarnet_core log)
+  target_link_libraries(${husarnet_core} log)
 endif()
 
-FetchContent_Declare(
-  libsodium
-  GIT_REPOSITORY https://github.com/jedisct1/libsodium.git
-  GIT_TAG 1.0.19
-)
-FetchContent_MakeAvailable(libsodium)
-
-# Generate compile-time version header
-# Needs to be populated manually from libsodium's configure.ac
-set(SODIUM_LIBRARY_VERSION_STRING "1.0.19")
-
-set(SODIUM_LIBRARY_VERSION_MAJOR 28)
-set(SODIUM_LIBRARY_VERSION_MINOR  0)
-include(${CMAKE_CURRENT_LIST_DIR}/lib-config/libsodium/generate.cmake)
+# Fetch and build libsodium
+# In ESP32 builds, libsodium is already provided by the ESP-IDF
+if (NOT DEFINED ESP_PLATFORM)
+  FetchContent_Declare(
+    libsodium
+    GIT_REPOSITORY https://github.com/jedisct1/libsodium.git
```

Comment:
It'd be nice to keep those declarations consistent - either all of them using GIT or HTTP